sn_transfers/src/cashnotes/cashnote.rs

```diff
@@ -187,13 +176,6 @@ impl CashNote {
         {
             return Err(TransferError::CashNoteCiphersNotPresentInTransactionOutput);
         }
-
-        // verify that all signed_spend reasons are equal
-        let spent_reason = self.spent_reason();
```

Comment:
Do we need to remove this now?

All parent spends for beta-reward forwarding _should_ be the same. 

So what are we losing here? (I know we aren't using `reason` at the moment, but the idea was to get this in so that we could (in theory) allow reupload of data that has already been paid for. 

That would rely upon parent spends being the same (the reason is the data addr), and royalty existing that matched amount, I think.

Reply:
The intention was indeed to reupload data that was paid for, but this feature couldn't work with batches properly so it was just forgotten. 
The reason I remove this check is that it protects us from virtually nothing and does not contribute to the security at all. 
Plus we agreed with Qi that spend reasons should be independent.

Reply:
Also, I removed the spent_reason method from CashNote as we discussed, making this check even more absurd